File: programmers.co.kr/hash/3.py
# https://programmers.co.kr/learn/courses/30/lessons/42578


def solution(clothes):
    d = {}
    answer = 1
    for __, clothingType in clothes:
        d[clothingType] = d.get(clothingType, 0) + 1

    for clothingType in d.keys():
        answer *= (d[clothingType] + 1)
    answer -= 1
    return answer

File: programmers.co.kr/stacks_queues/3.cpp
// https://programmers.co.kr/learn/courses/30/lessons/42583
// Trucks Crossing a Bridge

#include <queue>
#include <string>
#include <vector>

using namespace std;

int solution(int bridge_length, int weight, vector<int> truck_weights) {
    int answer = 0;
    queue<int> q;
    int index = 0;
    int sum = 0;

    while (index != truck_weights.size()) {
        answer++;

        if (q.size() == bridge_length) {
            sum -= q.front();
            q.pop();
        }

        int next = truck_weights[index];
        if (sum + next <= weight) {
            q.push(next);
            sum += next;
            index++;
        } else {
            q.push(0);
        }
    }

    answer += bridge_length;

    return answer;
}
